backend/app/services/encryption.py

```
@@ -0,0 +1,141 @@
+        key_str = provided_key or os.environ.get("ENCRYPTION_KEY")
+        if key_str:
+            return base64.b64decode(key_str)
+        debug = os.environ.get("DEBUG", "true").lower() == "true"
+        if debug:
+            logger.warning(
+                "No ENCRYPTION_KEY set. Generating temporary key for development."
+            )
+            return secrets.token_bytes(32)
+        raise EncryptionError(
+            "ENCRYPTION_KEY environment variable required in production"
+        )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)